src/v/cloud_storage/recursive_directory_walker.h

```diff
@@ -29,6 +29,7 @@ struct file_list_item {
 struct walk_result {
     uint64_t cache_size{0};
     std::vector<file_list_item> deletion_candidates;
+    std::vector<ss::sstring> empty_dirs;
```

Comment:
nit: I was a bit confused by the naming initially as `empty_dirs` are also "candidates for deletion", but I guess the `file_list_item` indicates the fact that `deletion_candidates` are files, which justifies the different treatment.